Core/src/main/java/org/tribuo/onnx/ONNXAttribute.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.tribuo.onnx;
+
+import com.google.protobuf.ByteString;
+import ai.onnx.proto.OnnxMl;
+
+import java.nio.charset.StandardCharsets;
+
+/**
+ * The spec for an attribute, used to produce the attribute proto at construction time.
+ */
+public final class ONNXAttribute {
+
+    private final String name;
+    private final OnnxMl.AttributeProto.AttributeType type;
+
+    /**
+     * Constructs an attribute placeholder of the appropriate name and type.
+     *
+     * @param name The name of the attribute.
+     * @param type The type of the attribute.
+     */
+    public ONNXAttribute(String name, OnnxMl.AttributeProto.AttributeType type) {
+        this.name = name;
+        this.type = type;
+    }
+
+    /**
+     * Gets the attribute's name.
+     *
+     * @return The attribute's name.
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Gets the attribute's type.
+     *
+     * @return The attribute's type.
+     */
+    public OnnxMl.AttributeProto.AttributeType getType() {
+        return type;
+    }
+
+    /**
+     * Builds the attribute proto using the supplied value.
+     * <p>
+     * Throws {@link IllegalArgumentException} if the value type does not match the expected type, and
+     * throws {@link UnsupportedOperationException} if the value type is not supported. Currently supported
+     * types are primitives, strings and arrays of primitives and strings.
```

Comment:
How do `long` and `double` work here? Are they unsupported by ONNX?

Reply:
I think I could set them as `Tensor` which supports that. However this appears to be how pytorch encodes the values, as the protobufs I was generating are very similar.

In general ONNX's support of `double` is pretty lacking, to the point where I only bother to support `float` in Tribuo's ONNX Runtime interface, and I downcast everything to `float` on export.

Reply:
Ok, in that case maybe we should document that `long` and `double` will be downcast then, so that when it happens people will be less surprised.

Reply:
Sure.